### PR TITLE
feat!: add native plugin example

### PR DESCRIPTION
**Pull request by csmith0651**
### Description

* Update to look, feel, and smell like the angular my crasher demo.
* Add native plugin (addon) example.

### CODE REVIEW

**Comment by bobbyg603** [comment]
UI looks great! I'll add feedback to this comment as I find it. 

Here's what I have so far:

1. Can we make the links in the app open in the user's default browser (outside of the app)?
2. Match the my-angular-crasher border style (no left-right border, small light gray top border)
3. There's a few pieces of commented/dead code, let's make a pass at removing some of that before merging

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
 node_modules
 dist
 .DS_Store
-.env
\ No newline at end of file
+.env
+/build/Debug/addon.node
+/build
```
Comment (bobbyg603) [inline .gitignore:5]: can we just exclude all of `/build`?

File: index.html [hunk 2]
```diff
@@ -1,45 +1,46 @@
...
+                    <button class="error" id="native-crash-button">
+                        <h2>Native (plugin) Crash</h2>
+                        <h4>dereference null pointer in native code</h4>
+                        <h4>WILL TERMINATE APP</h4>
+                    </button>
+                </div>
+            </div>
+        </div>
+        <script src="./dist/src/renderer.js"></script>
+    </body>
```
Comment (bobbyg603) [inline index.html:45]: Very clean, nicely done!

File: index.html [hunk 3]
```diff
@@ -1,45 +1,46 @@
...
-</body>
-
+            <div class="content">
+                <h1>
+                    Welcome to my-electron-crasher
+                </h1>
+                <p>
+                    This is a sample application that demonstrates <a href="www.bugsplat.com">BugSplat</a> error
+                    reporting
+                    for <a href="www.electronjs.org">Electron</a> applications build with JavaScript or TypeScript.
```
Comment (bobbyg603) [inline index.html]: Let's see if we can get links to open the user's default browser (outside of the app)

File: src/assets/css/styles.css [hunk 4]
```diff
@@ -1,6 +1,87 @@
 .container-fluid {
-    background-color: #00386c;
+    /*background-color: #00386c;*/
     color: #ffffff;
     display: block;
     height: 100vh;
-}
\ No newline at end of file
+    background-image: url('../img/background.png');
+}
```
Comment (bobbyg603) [inline src/assets/css/styles.css]: It looks like `.container-fluid` is no longer used, can we remove these styles?

File: src/assets/css/styles.css [hunk 5]
```diff
@@ -1,6 +1,87 @@
...
+
+h1,
+h2 {
+    color: #007bff;
+}
+
+
```
Comment (bobbyg603) [inline src/assets/css/styles.css:10]:
```suggestion
```
nit: remove extra space

File: src/assets/css/styles.css [hunk 6]
```diff
@@ -1,6 +1,87 @@
...
+    transition-property: background-color;
+}
+
+/* button, with class of error, and when hover is triggered */
+button.error:hover {
+    background: rgb(245, 245, 245);
+}
+
+/* button, with class of error, when focus is triggered */
+
```
Comment (bobbyg603) [inline src/assets/css/styles.css:70]:
```suggestion
```
nit: remove extra space

File: src/main.ts [hunk 7]
```diff
@@ -84,15 +86,12 @@ app.on("window-all-closed", () => {
   }
 });
 
-// Called from renderer.js to test JavaScript error handling
-ipcMain.on("mainError", function () {
-  uncaughtException()
+ipcMain.on("trigger:user-main-crash", function () {
+  console.log(`on.trigger:user-main-crash`);
+  unhandledRejection("unhandledRejection: main process");
+  // ipcUserMainCrash();
```
Comment (bobbyg603) [inline src/main.ts]: do we still need this commented code?

File: src/renderer.ts [hunk 8]
```diff
@@ -27,22 +27,24 @@ window.onerror = async (messageOrEvent, source, lineno, colno, error) => {
...
 
 window.onload = function () {
-    document.getElementById("main-error-button-container").innerHTML = "<button id=\"main-error-button\" class=\"btn btn-danger\">JavaScript (Main) Error</button>";
-    document.getElementById("renderer-error-button-container").innerHTML = "<button id=\"renderer-error-button\" class=\"btn btn-danger\">JavaScript (Renderer) Error</button>"
-    document.getElementById("native-crash-button-container").innerHTML = "<button id=\"native-crash-button\" class=\"btn btn-danger\">Native Crash</button>"
+    console.log('in onload');
+
+    // document.getElementById("main-error-button-container").innerHTML = "<button id=\"main-error-button\" class=\"btn btn-danger\">JavaScript (Main) Error</button>";
+    // document.getElementById("renderer-error-button-container").innerHTML = "<button id=\"renderer-error-button\" class=\"btn btn-danger\">JavaScript (Renderer) Error</button>"
+    // document.getElementById("native-error-button-container").innerHTML = "<button id=\"native-crash-button\" class=\"btn btn-danger\">Native Crash</button>"
```
Comment (bobbyg603) [inline src/renderer.ts]: Can we remove this stuff?

File: src/assets/css/styles.css (hunk 4 above)
Comment (csmith0651) [reply src/assets/css/styles.css]: fixed

File: src/main.ts (hunk 7 above)
Comment (csmith0651) [reply src/main.ts]: nope, removed.

File: src/renderer.ts (hunk 8 above)
Comment (csmith0651) [reply src/renderer.ts]: yup. removed.

**Review by bobbyg603 — APPROVED** [review]

File: index.html (hunk 3 above)
Comment (csmith0651) [reply index.html]: fixed.

**Comment by bobbyg603** [comment]
Thanks for the contribution @csmith0651 🥳🍻🤩

**Pull request merged**